test/back_end/api_post_test.spec.js

```diff
@@ -110,6 +110,49 @@ describe("Test POST/UPDATE/DELETE requests to API", function() {
             }
         });
 
+        it("launch the survey with survey_id = 1", async function () {
+            try {
+                let fetched = await fetch(
+                    "http://localhost:3000/post_info.php?what=launch_survey&user_id=woods13&survey_id=1&action=add_or_update"
+                );
+                expect(fetched).to.have.status(200);
+                let fetchedJSON = await fetched.json();
+                expect(fetchedJSON.TYPE).to.be.equal("survey_package");
+                expect(parseInt(fetchedJSON.DATA.survey_instance_id)).to.satisfy(Number.isInteger);
+                let survey_instance_id = fetchedJSON.DATA.survey_instance_id;
+                let override_token = fetchedJSON.DATA.override_token;
+                fetched = await fetch(
+                    "http://localhost:3000/get_info.php?what=survey_results&user_id=woods13&course=UofT&target_ta=woods13&term=201709"
```

Comment:
Right now `launch_survey` dynamically computes the `term`. I don't think your get request is robust...